plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/MultipathSCSIAdapterBase.java

```diff
@@ -223,15 +228,19 @@ public boolean disconnectPhysicalDisk(Map<String, String> volumeToDisconnect) {
     @Override
     public boolean disconnectPhysicalDiskByPath(String localPath) {
         LOGGER.debug(String.format("disconnectPhysicalDiskByPath(localPath) called with args (%s) STARTED", localPath));
+        if (localPath == null || (localPath != null && !localPath.startsWith("/dev/mapper/"))) {
```

Comment:
Unfortunately, there is no other distinguishing factor I can think of in this scenario to determine if the MultipathSCSIAdapter code should take action or not.  This flow is called when the LibvirtComputingResource cleans up disks left over after a live migration (from what I can tell) and it only knows the path that was referenced in the Domain configuration (the multipathed file).  /dev/mapper is the standard location for multipath mapped devices.  Even if another storage provider uses this path, this cleanup routine should not cause an issue since it could logically only perform the same function (delete the multipath and clean up any residual path devices left).

Reply:
Agreed it is a shortcoming.  I am looking at one option to potentially evaluate the path before deciding whether we should process it if the path is a WWID-based path and if the OUI matches one that is known to the driver (Primera or Pure).

Reply:
I added logic as noted and this should no longer conflict with another storage plugin using multipath.